src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  return actions.reduce((acc, action, index) => {
+    const statePrevious = index === 0 ? { ...state } : { ...acc[index - 1] };
```

Comment:
The naming of the object copy could be improved to reflect its purpose more clearly. Instead of `statePrevious`, a name like `stateCopy` would be more in line with the [NAMING] best practices.